src/Microsoft.AspNetCore.Rewrite/Logging/RewriteMiddlewareLoggingExtensions.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Rewrite.Logging
+{
+    internal static class RewriteMiddlewareLoggingExtensions
+    {
+        private static readonly Action<ILogger, Exception> _requestContinueResults;
+        private static readonly Action<ILogger, string, Exception> _requestResponseComplete;
+        private static readonly Action<ILogger, Exception> _requestStopRules;
+        private static readonly Action<ILogger, string, Exception> _urlRewriteDidNotMatchRule;
+        private static readonly Action<ILogger, string, Exception> _urlRewriteMatchedRule;
+        private static readonly Action<ILogger, Exception> _modRewriteDidNotMatchRule;
+        private static readonly Action<ILogger, Exception> _modRewriteMatchedRule;
+
+        static RewriteMiddlewareLoggingExtensions()
+        {
+            _requestContinueResults = LoggerMessage.Define(
+                            LogLevel.Debug,
+                            1,
+                            "Request is continuing in applying rules.");
+            _requestResponseComplete = LoggerMessage.Define<string>(
+                            LogLevel.Debug,
+                            2,
+                            "Request is done processing, Location header '{Location}'.");
```

Comment:
Showing the status code would be more generalized, not all response completers set Location